bd-api/src/api.rs

```diff
@@ -830,11 +830,15 @@ impl Api {
             .upload_state_tracker
             .resolve_pending_upload(&stats_upload.upload_uuid, &stats_upload.error)?;
         },
-        Some(ResponseKind::FlushBuffers(flush_buffers)) => self
-          .trigger_upload_tx
-          .send(TriggerUpload::new(flush_buffers.buffer_id_list.clone()))
-          .await
-          .map_err(|_| anyhow!("remote trigger upload tx"))?,
+        Some(ResponseKind::FlushBuffers(flush_buffers)) => {
+          let (tx, _rx) = tokio::sync::oneshot::channel();
+
+          self
+            .trigger_upload_tx
+            .send(TriggerUpload::new(flush_buffers.buffer_id_list.clone(), tx))
+            .await
+            .map_err(|_| anyhow!("remote trigger upload tx"))?;
```

Comment:
One issue is that it would bypass all the logic that I added here and we'd get dropped logs as the workflow engine thinks its safe to write to the trigger buffer. This is no different from how this used to work when we would just lock the buffer for upload, but something to keep in mind.

I think if we're to use this in any serious manner we'd want to either have this flush trigger interact with the workflow engine or just facilitate this via some temporary workflow that is used to trigger a flush on next log as was mentioned in a previous meeting